astropy/utils/masked/core.py

```diff
@@ -769,7 +793,11 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
 
         elif method == "__call__":
             # Regular ufunc call.
-            mask = self._combine_masks(masks, out=out_mask)
+            # Combine the masks from the input, possibly selecting elements.
+            mask = self._combine_masks(masks, out=out_mask, where=where_unmasked)
+            # If relevant, also mask output elements for which where was masked.
+            if where_mask is not None:
+                mask |= where_mask
```

Comment:
Would it be clearer to do something like
```suggestion
            mask = self._combine_masks(masks + [where_mask], out=out_mask, where=where_unmasked)
```

Reply:
I like the green CI but I should probably wait till this comment is resolved before approving?